machine/v1/types_alibabaprovider.go

```diff
@@ -153,13 +161,16 @@ type AlibabaCloudMachineProviderConfig struct {
 // Only one of ID or Tags may be specified. Specifying more than one will result in
 // a validation error.
 type AlibabaResourceReference struct {
+	// type identifies the resource reference type for this entry.
+	Type AlibabaResourceReferenceType `json:"type"`
+
 	// ID of resource
 	// +optional
-	ID string `json:"id,omitempty"`
+	ID *string `json:"id,omitempty"`
 
 	// Tags is a set of metadata based upon ECS object tags used to identify a resource
```

Comment:
can you clarify what happens if Tags produces multiple matches?

Reply:
will do